ZupIT/beagle-web-angular
# Removes the markdown component and other dependencies

## What I did
closes https://github.com/ZupIT/beagle-web-core/issues/288

I reduced the size of the Beagle Angular library by removing the dependency `showdown`.

## How I did it
### Showdown
I removed the component `markdown` so that the dependency `showdown` could be removed. This dependency is way too big and this component probably won't ever be used. There's no reason to increase the size of Beagle only because of this component.

Since the component `markdown` is required in the Beagle Playground, it has been transferred there. Please, also accept [this PR](https://github.com/ZupIT/beagle-playground/pull/72).

### Result
I created three apps with ng new (Angular 9). One without Beagle, another with Beagle 1.3.0 and another with Beagle at this branch (fix/lib-size). I compared the sizes of each generated bundle (production) using 4 different methods:

- **directory size:** raw size of the bundle directory
- **initial payload:** when opening the application in a browser, how much data it needs to download? The value were taken from the network tab of Chrome dev tools.
- **webpack bundle analyzer:** I used [this tool](https://www.npmjs.com/package/webpack-bundle-analyzer) to measure the bundle size. I used the measuring "Parsed" to get these results, which is the size of the minified files.
- **webpack bundle analyzer (gzip):** I used [this tool](https://www.npmjs.com/package/webpack-bundle-analyzer) to measure the bundle size. I used the measuring "Gzipped" to get these results, which is the size of the compressed minified files.

|                                | Without Beagle | With Beagle 1.3.0 | With Beagle at this branch |
|--------------------------------|----------------|-------------------|----------------------------|
| Directory size                 | 639KB          | 1.4MB             | 1.2MB                      |
| Initial payload                | 231KB          | 583KB             | 497KB                      |
| Webpack bundle analyzer        | 353KB          | 697KB             | 613KB                      |
| Webpack bundle analyzer (gzip) | 110KB          | 201KB             | 175KB                      |

## Related Pull Requests
- Beagle React: https://github.com/ZupIT/beagle-web-react/pull/159
- Playground: https://github.com/ZupIT/beagle-playground/pull/72

## How to verify it
To verify the sizes, you must compare the sizes of the same application with and without beagle.
To verify everything keeps working as intended, you can visit the [playground beta](https://beagle-playground-beta.netlify.app/), which represents this exact branch.

## Description for the changelog
- removed the component `markdown` and the dependency `showdown` so beagle-angular can have a smaller footprint.
